source/fundamentals/data-formats/bson.txt

```diff
@@ -0,0 +1,241 @@
+.. _csharp-bson:
+
+==============
+Work with BSON
+==============
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to create BSON documents, read BSON from a file, 
+and write BSON to a file using the {+driver-short+}.
+
+BSON Data Format
+----------------
+
+**BSON**, or Binary JSON, is the data format that MongoDB uses to organize
+and store data. This data format includes all JSON data structure types and
+adds support for types including dates, different size integers, ObjectIds, and
+binary data. For a complete list of supported types, see the
+:manual:`BSON Types </reference/bson-types>` server manual page.
+
+The code samples in this guide use the following BSON document as an example:
+
+.. code-block:: none
+
+   {
+       "address" : {
+           "street" : "Pizza St", 
+           "zipcode" : "10003" 
+       },
+       "coord" : [-73.982419, 41.579505]
+       "cuisine" : "Pizza",
+       "name" : "Mongo's Pizza"
+   }
+
+Create a BSON Document
+----------------------
+
+To build a BSON document in {+language+}, create an instance of the ``BsonDocument`` class.
+The ``BsonDocument`` constructor accepts ``BsonElement`` arguments that map to the fields
+and values in the document. Each ``BsonElement`` can be either an instance of the 
+``BsonElement`` class or a field-value pair inside curly braces ( ``{}`` ).
+
+The following code sample shows how to create a ``BsonDocument`` object to represent the 
+example BSON document. Each ``BsonElement`` is an instance of the ``BsonElement``
+class.
+
+.. literalinclude:: ../../includes/fundamentals/code-examples/Bson.cs
+   :start-after: start-bson-elements
+   :end-before: end-bson-elements
+   :language: csharp 
+   :copyable:
+   :dedent:
+
+The following code sample creates the same ``BsonDocument``, but each ``BsonElement``
+is a field-value pair: 
```

Comment:
I think the intention here is just to make sure the reader is aware that there are two different ways to specify the same document - in case they come across it at some point. If it's not worth mentioning though I think we could probably just remove it.
@mongoKart feel free to correct me if I'm wrong on my reasoning here

Reply:
That's right; users might see both syntaxes, and I also thought showing both sequentially would help them make the connection between `BsonElement` and a key-value pair. 

I'm in favor of showing the more idiomatic form and removing the `BsonElement` example.

Reply:
The dictionary initialization syntax is common in .NET. (e.g. `new SomeDict {{ key1, value1 }}`) Internally `new SomeDict {{ key1, value1}}` actually creates `KeyValuePair` objects just as we create `BsonElement` objects. Thus I think that `new BsonDocument {{ key1, value1 }}` should be obvious to most C# developers.

As for the compilation error, the `BsonArray` is initialized as:

```
new BsonArray {{ value1, value2 }}
```

But that is dictionary initialization syntax (taking key/value pairs), not array initialization syntax (taking a list of items). It should be:

```
new BsonArray { value1, value2 }
```

Reply:
Makes sense. Let me know if my changes address these comments.